vm/ubpf_vm.c

```diff
@@ -573,6 +573,27 @@ ubpf_exec(const struct ubpf_vm *vm, void *mem, size_t mem_len, uint64_t* bpf_ret
     }
 }
 
+uint64_t
+ubpf_run(struct ubpf_vm *vm, void *mem, size_t mem_len, bool jit)
+{
+    uint64_t ret;
+    char *errmsg;
+
+    if (jit) {
+	ubpf_jit_fn fn = ubpf_compile(vm, &errmsg);
+	if (fn == NULL) {
+	    fprintf(stderr, "Failed to compile: %s\n", errmsg);
```

Comment:
It's generally bad form for library functions to print their own error messages.... This helper should probably return it's own errmsg using the usual pattern.